### PR TITLE
[sailfishos][embedlite] Create BrowserChildHelper early enough. JB#55336 OMP#JOLLA-331

### PR DESCRIPTION
**Pull request by rainemak**
This PR contains three commits. One fixing EmbedLiteViewChild::InitGeckoWindow, hand picking and adapting changes from the https://github.com/sailfishos/gecko-dev/commit/bbd864aca040a90e72a6e4a9a378c741d0dcf1d6 (0020-sailfishos-loginmanager-Adapt-LoginManager-to-EmbedL.patch) and make it possible to get ContentFrameMessageManager via nsDocShell.

nsIBrowserChild (BrowserChildHelper) implementation is available
for nsIWebBrowserChrome (WebBrowserChrome). In follow up patch,
ContentFrameMessageManager is retrieved via nsDocShell::SetTreeOwner.

Make it possible to take a weak reference to the nsIBrowserChild.

This requires a small patch to the nsDocShell (see rpm directory).

SetContainerWindow is set already by the nsWebBrowser::Create.

### CODE REVIEW

**Review by mkenttala — APPROVED** [review]

**Comment by rainemak** [comment]
Rebased

**Review by paveltkv — APPROVED** [review]

File: rpm/xulrunner-qt5.spec [hunk 1]
```diff
@@ -88,6 +88,8 @@ Patch39:    0039-sailfishos-embedlite-egl-Fix-mesa-egl-display-initia.patch
 Patch40:    0040-sailfishos-egl-Do-not-create-CreateFallbackSurface.-.patch
 Patch41:    0041-sailfishos-gecko-Fix-gfxPlatform-AsyncPanZoomEnabled.patch
 Patch42:    0042-sailfishos-gecko-Avoid-incorrect-compiler-optimisati.patch
+Patch43:    0043-sailfishos-gecko-Drop-static-casting-from-BrowserChi.patch
```
Comment (okodron) [inline rpm/xulrunner-qt5.spec:91]: However the patch name is`0042-sailfishos-gecko-Drop-static-casting-from-BrowserChi.patch`. Rebase issue?

Comment (paveltkv) [reply rpm/xulrunner-qt5.spec:91]: Also 0043-sailfishos-gecko-Avoid-incorrect-compiler-optimisati.patch doesn't match

Comment (rainemak) [reply rpm/xulrunner-qt5.spec:91]: Amended commit and renamed rpm/0042-sailfishos-gecko-Avoid-incorrect-compiler-optimisati.patch to rpm/0043-sailfishos-gecko-Drop-static-casting-from-BrowserChi.patch

**Comment by rainemak** [comment]
Still https://github.com/sailfishos/gecko-dev/pull/37/files shows "...Avoid-incorrect-compiler-optimisati.patch → ...Avoid-incorrect-compiler-optimisati.patch". Hmm

**Review by okodron — APPROVED** [review]

**Comment by rainemak** [comment]
> Still https://github.com/sailfishos/gecko-dev/pull/37/files shows "...Avoid-incorrect-compiler-optimisati.patch → ...Avoid-incorrect-compiler-optimisati.patch". Hmm

Corrected this as well.

**Pull request merged**